clients/privacy-center/components/modals/consent-request-modal/ConsentRequestForm.tsx

```diff
@@ -254,7 +257,7 @@ const ConsentRequestForm: React.FC<ConsentRequestFormProps> = ({
             _active={{ bg: "primary.500" }}
             colorScheme="primary"
             isLoading={isSubmitting}
-            isDisabled={isSubmitting || !(isValid && dirty)}
+            // isDisabled={isSubmitting || !(isValid && dirty)}
```

Comment:
We'll need to switch this to selectively disable the button only when no field inside `config.consent.button.identity_inputs` is `"required"`

Reply:
👍 we can do that through the `validationSchema`, i.e. pass a different schema depending on how the config looks. let me know if that makes sense or if you'd like to pair on it!

Reply:
I looked into this a bit and there's already logic in `validationSchema` which only makes the field required if the config says to, so I think we are good there.

it looks like the problem is actually that the form isn't dirty—we make it so you can't submit the form if it isn't dirty. so you can type something into the email form and then erase it and then that actually lets you submit, which likely isn't what we want.

I think we could change the `dirty` check such that it's only required to be dirty if there is a required field

```typescript
  // figure out all inputs that are required
  const requiredInputs = Object.entries(identityInputs).filter(
    ([, required]) => required === "required"
  );
  // it's ok to bypass the dirty check if there are no required inputs
  const dirtyCheck = requiredInputs.length === 0 ? true : dirty;

  ...
  // in <Button />
  isDisabled={isSubmitting || !(isValid && dirtyCheck)}
```

that said, that UX feels a little weird, to just bypass the email field